packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -15,40 +15,107 @@
+ * code that does so.
+ */
+export class AggregateQuery {
   readonly type = 'AggregateQuery';
-  readonly query: Query<T>;
+  /**
+   * The query on which you called {@link countQuery} in order to get this
+   * `AggregateQuery`.
+   * Query type is set to unknown to avoid error caused by query type converter.
+   * might change it back to T after testing if the error do exist or not
+   */
+  readonly query: Query<unknown>;
 
   /** @hideconstructor */
-  constructor(query: Query<T>) {
+  constructor(query: Query<unknown>) {
     this.query = query;
   }
+
 }
 
+/**
+ * A {@code AggregateQuerySnapshot} contains results of a {@link AggregateQuery}.
+ *
+ * <p><b>Subclassing Note</b>: Cloud Firestore classes are not meant to be subclassed except for use
+ * in test mocks. Subclassing is not supported in production code and new SDK releases may break
+ * code that does so.
+ */
 export class AggregateQuerySnapshot {
   readonly type = 'AggregateQuerySnapshot';
   readonly query: AggregateQuery;
 
   /** @hideconstructor */
-  constructor(query: AggregateQuery, readonly _count: number) {
+  constructor(query: AggregateQuery, private readonly _count: number) {
     this.query = query;
   }
 
+  /**
+   * @return The result of a document count aggregation. Returns null if no count aggregation is
+   * available in the result.
+   */
   getCount(): number | null {
     return this._count;
   }
 }
 
-export function countQuery(query: Query): AggregateQuery {
+/**
+ * Creates an {@link AggregateQuery} counting the number of documents matching this query.
+ *
+ * @return An {@link AggregateQuery} object that can be used to count the number of documents in
+ * the result set of this query.
+ */
+export function countQuery(query: Query<unknown>): AggregateQuery {
+  /**
+   * TODO(mila): add the "count" aggregateField to the params after the AggregateQuery is updated.
+   */
   return new AggregateQuery(query);
 }
 
 export function getAggregateFromServerDirect(
-  query: AggregateQuery
+  aggregateQuery: AggregateQuery
 ): Promise<AggregateQuerySnapshot> {
-  return Promise.resolve(new AggregateQuerySnapshot(query, 42));
+  const firestore = cast(aggregateQuery.query.firestore, Firestore);
+  const datastore = getDatastore(firestore);
+  const userDataWriter = new LiteUserDataWriter(firestore);
+
+  return invokeRunAggregationQueryRpc(datastore, aggregateQuery).then(
+    result => {
+      const aggregationFields = new Map<string, any>();
+      /**
+       * while getting aggregation fields from server direct, it should have only
+       * one RunAggregationQueryResponse returned.
+       * But we used streaming rpc here, so we will have an array of
+       * (one, or possibly more) RunAggregationQueryResponse. For this specific
+       * function, we get the first RunAggregationQueryResponse only.
+       */
+      for (const [key, value] of Object.entries(result[0])) {
+        aggregationFields.set(key, userDataWriter.convertValue(value));
+      }
```

Comment:
Since you know what you're looking for, just access `result[0].count_alias`. Use `hardAssert()` to fail if it is `=== undefined`. Then, use `hardAssert()` to fail if the value's type is not numeric. Finally, return the resolved promise with the count result.

Reply:
With Value type, I cannot read result[0].count_alias directly, so I used a loop to project the result[0] into a map object, then read the count_alias field.
![Screen Shot 2022-08-23 at 9 30 41 AM](https://user-images.githubusercontent.com/107142260/186212601-888dc933-241e-4851-8d32-ecf1525d1013.png)